src/Setup/WixBA/InstallationViewModel.cs

```diff
@@ -502,15 +452,21 @@ private void CacheAcquireComplete(object sender, CacheAcquireCompleteEventArgs e
 
         private void ExecutePackageBegin(object sender, ExecutePackageBeginEventArgs e)
         {
-            this.executePackageStart = e.ShouldExecute ? DateTime.Now : DateTime.MinValue;
+            lock (this)
```

Comment:
From gmail I wrote a response to this, but since I don't see it, and at the risk of duplicating, let me state my concern with removing the lock().  As I understand it in WPF each ViewModel is on a separate thread, and long ago when my mba was having some issues, I noticed that the WixBA (3.8?) used lock() when more than one VM provides a handler for the same event.  Adding lock() in that scenario resolved my problems, though I do not recall the exact details.  IF you want me to remove the lock() I can do so, but given that each VM is a separate thread, doesn't using the lock() serve a valid purpose here?  (Please take this as a comment for each of the places where a concern was raised about using lock(). )